torchrl/envs/common.py

```diff
@@ -218,10 +228,22 @@ def __init__(
             self.batch_size = torch.Size([])
 
     @classmethod
-    def __new__(cls, *args, **kwargs):
+    def __new__(cls, *args, _batch_locked=True, **kwargs):
         cls._inplace_update = True
+        cls._batch_locked = _batch_locked
         return super().__new__(cls)
 
+    @property
+    def batch_locked(self) -> bool:
+        """
+        Whether the environnement can be used with a batch size different from the one it was initialized with or not.
```

Comment:
"""Whether the environnement can be used with a batch size different from the one it was initialised with."""

Also, I would add that this is an immutable property. 

And we don't get if `True` means that the batch size can be different of not. The proper formulation would be "returns True if the env blabla"

Can you provide an example in the docstring?